File: check_submission_detailed.py
#!/usr/bin/env python3
"""
Detailed check of a submission including all SEO and AI fields
"""
import os
import sys
import json
from datetime import datetime
from supabase import create_client, Client
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

# Initialize Supabase client
url = os.environ.get("VITE_SUPABASE_URL")
key = os.environ.get("VITE_SUPABASE_ANON_KEY")

if not url or not key:
    print("Error: Missing Supabase credentials")
    exit(1)

supabase: Client = create_client(url, key)

# Get submission ID
submission_id = "367789ee-9e5d-4a16-9c27-30d475736dab"

print(f"Detailed check for submission: {submission_id}\n")

# Fetch submission with all fields
result = supabase.table('submissions').select("*").eq('id', submission_id).execute()

if result.data:
    submission = result.data[0]
    
    print("=== AI PROCESSING STATUS ===")
    print(f"ai_processing_status: {submission.get('ai_processing_status')}")
    print(f"ai_generated_content: {'YES' if submission.get('ai_generated_content') else 'NO'}")
    print(f"ai_error: {submission.get('ai_error')}")
    
    print("\n=== SEO CONTENT FIELDS ===")
    print(f"seo_content: {'YES' if submission.get('seo_content') else 'NO'}")
    if submission.get('seo_content'):
        print(f"  Length: {len(submission.get('seo_content', ''))} characters")
        print(f"  Preview: {submission.get('seo_content', '')[:200]}...")
    
    print(f"\nseo_title: {submission.get('seo_title')}")
    print(f"meta_title: {submission.get('meta_title')}")
    print(f"meta_description: {submission.get('meta_description')}")
    print(f"seo_keywords: {submission.get('seo_keywords')}")
    
    print("\n=== OTHER SEO FIELDS ===")
    print(f"primary_keywords: {submission.get('primary_keywords')}")
    print(f"secondary_keywords: {submission.get('secondary_keywords')}")
    print(f"h1_tag: {submission.get('h1_tag')}")
    print(f"h2_tags: {submission.get('h2_tags')}")
    print(f"geo_event_tags: {submission.get('geo_event_tags')}")
    print(f"seo_strategy_outline: {'YES' if submission.get('seo_strategy_outline') else 'NO'}")
    print(f"competitive_analysis: {'YES' if submission.get('competitive_analysis') else 'NO'}")
    
    print("\n=== TIMESTAMPS ===")
    print(f"created_at: {submission.get('created_at')}")
    print(f"updated_at: {submission.get('updated_at')}")
    print(f"ai_processing_started_at: {submission.get('ai_processing_started_at')}")
    print(f"ai_processing_completed_at: {submission.get('ai_processing_completed_at')}")
    
    print("\n=== WORKFLOW INFO ===")
    print(f"workflow_stage: {submission.get('workflow_stage')}")
    print(f"status: {submission.get('status')}")
    
    # Check if ai_generated_content has any content
    if submission.get('ai_generated_content'):
        print("\n=== AI GENERATED CONTENT DETAILS ===")
        try:
            # Try to parse if it's JSON
            ai_content = json.loads(submission.get('ai_generated_content'))
            print("Content is JSON format:")
            for key, value in ai_content.items():
                print(f"  {key}: {str(value)[:100]}...")
        except:
            # If not JSON, show as string
            content = submission.get('ai_generated_content')
            print(f"Content is text format ({len(content)} characters)")
            print(f"Preview: {content[:500]}...")
    
    # Save complete results
    output_file = f"detailed_submission_{submission_id}.json"
    with open(output_file, 'w') as f:
        json.dump(submission, f, indent=2)
    print(f"\n\nComplete submission data saved to: {output_file}")
    
else:
    print(f"❌ Submission not found: {submission_id}")

File: check-submission-detailed.mjs
import { createClient } from '@supabase/supabase-js';
import dotenv from 'dotenv';

dotenv.config();

const supabase = createClient(
  process.env.VITE_SUPABASE_URL,
  process.env.VITE_SUPABASE_ANON_KEY
);

async function checkSubmissionDetailed() {
  const submissionId = '695e8b9d-9e52-4288-8d2b-6551fae55375';
  
  console.log(`Checking submission ID: ${submissionId}\n`);

  // Fetch ALL fields from the submission
  let { data: submission, error } = await supabase
    .from('submissions')
    .select('*')
    .eq('id', submissionId)
    .single();

  if (error) {
    console.error('Error fetching submission:', error);
    return;
  }

  if (!submission) {
    console.log('No submission found with this ID');
    return;
  }

  console.log('=== BASIC INFORMATION ===\n');
  console.log(`ID: ${submission.id}`);
  console.log(`Compliance ID: ${submission.compliance_id || 'N/A'}`);
  console.log(`Product Name: ${submission.product_name || 'N/A'}`);
  console.log(`Submitter: ${submission.submitter_name} (${submission.submitter_email})`);
  console.log(`Company: ${submission.submitter_company || 'N/A'}`);
  console.log(`Created: ${submission.created_at}`);
  console.log(`Last Updated: ${submission.updated_at}`);

  console.log('\n=== WORKFLOW STATUS ===\n');
  console.log(`Workflow Stage: ${submission.workflow_stage || 'Not set'}`);
  console.log(`AI Processing Status: ${submission.ai_processing_status || 'Not set'}`);
  console.log(`AI Phase: ${submission.ai_phase || 'Not set'}`);
  console.log(`AI Status: ${submission.ai_status || 'Not set'}`);
  console.log(`AI Retry Count: ${submission.ai_retry_count || 0}`);
  console.log(`AI Last Retry: ${submission.ai_last_retry || 'N/A'}`);

  console.log('\n=== CONTENT STATUS ===\n');
  
  // Raw Input Content
  console.log('1. Raw Input Content:');
  if (submission.raw_input_content) {
    console.log(`   ✅ POPULATED (${submission.raw_input_content.length} characters)`);
    console.log(`   Preview: ${submission.raw_input_content.substring(0, 200)}...`);
  } else {
    console.log('   ❌ EMPTY');
  }

  // AI Generated Content
  console.log('\n2. AI Generated Content:');
  if (submission.ai_generated_content) {
    console.log(`   ✅ POPULATED`);
    if (typeof submission.ai_generated_content === 'string') {
      console.log(`   Length: ${submission.ai_generated_content.length} characters`);
      console.log(`   Preview: ${submission.ai_generated_content.substring(0, 200)}...`);
    } else {
      console.log(`   Type: ${typeof submission.ai_generated_content}`);
      console.log(`   Content:`, JSON.stringify(submission.ai_generated_content, null, 2));
    }
  } else {
    console.log('   ❌ EMPTY');
  }

  // AI Output
  console.log('\n3. AI Output:');
  if (submission.ai_output) {
    console.log(`   ✅ POPULATED`);
    console.log(`   Type: ${typeof submission.ai_output}`);
    if (typeof submission.ai_output === 'object') {
      console.log(`   Keys: ${Object.keys(submission.ai_output).join(', ')}`);
    }
  } else {
    console.log('   ❌ EMPTY');
  }

  console.log('\n=== QA & FEEDBACK ===\n');
  
  // QA Feedback
  console.log('1. QA Feedback (Claude\'s Review):');
  if (submission.qa_feedback) {
    console.log(`   ✅ POPULATED`);
    if (typeof submission.qa_feedback === 'string') {
      console.log(`   Length: ${submission.qa_feedback.length} characters`);
      console.log(`   Preview: ${submission.qa_feedback.substring(0, 200)}...`);
    } else {
      console.log(`   Type: ${typeof submission.qa_feedback}`);
      console.log(`   Content:`, JSON.stringify(submission.qa_feedback, null, 2));
    }
  } else {
    console.log('   ❌ EMPTY');
  }

  // QA Score
  console.log('\n2. QA Score:');
  if (submission.qa_score !== null && submission.qa_score !== undefined) {
    console.log(`   ✅ Score: ${submission.qa_score}`);
  } else {
    console.log('   ❌ NOT SET');
  }

  console.log('\n=== SEO FIELDS ===\n');

  // SEO Keywords
  console.log('1. SEO Keywords:');
  if (submission.seo_keywords) {
    console.log(`   ✅ POPULATED`);
    if (Array.isArray(submission.seo_keywords)) {
      console.log(`   Keywords: ${submission.seo_keywords.join(', ')}`);
    } else {
      console.log(`   Value: ${submission.seo_keywords}`);
    }
  } else {
    console.log('   ❌ EMPTY');
  }

  // Meta Title
  console.log('\n2. Meta Title:');
  if (submission.meta_title) {
    console.log(`   ✅ POPULATED`);
    console.log(`   Title: ${submission.meta_title}`);
  } else {
    console.log('   ❌ EMPTY');
  }

  // Meta Description
  console.log('\n3. Meta Description:');
  if (submission.meta_description) {
    console.log(`   ✅ POPULATED`);
    console.log(`   Description: ${submission.meta_description}`);
  } else {
    console.log('   ❌ EMPTY');
  }

  // Additional SEO fields
  console.log('\n4. Additional SEO Fields:');
  console.log(`   Long Tail Keywords: ${submission.long_tail_keywords ? '✅' : '❌'}`);
  console.log(`   Consumer Questions: ${submission.consumer_questions ? '✅' : '❌'}`);
  console.log(`   H1 Tag: ${submission.h1_tag ? '✅' : '❌'}`);
  console.log(`   H2 Tags: ${submission.h2_tags ? '✅' : '❌'}`);
  console.log(`   SEO Title: ${submission.seo_title ? '✅' : '❌'}`);
  console.log(`   SEO Strategy Outline: ${submission.seo_strategy_outline ? '✅' : '❌'}`);

  console.log('\n=== ADDITIONAL INFORMATION ===\n');
  
  // Product details
  console.log('Product Information:');
  console.log(`   Therapeutic Area: ${submission.therapeutic_area || 'N/A'}`);
  console.log(`   Stage: ${submission.stage || 'N/A'}`);
  console.log(`   Priority Level: ${submission.priority_level || 'N/A'}`);
  console.log(`   Target Audience: ${submission.target_audience || 'N/A'}`);
  console.log(`   Medical Indication: ${submission.medical_indication || submission.indication || 'N/A'}`);
  console.log(`   Dosage Form: ${submission.dosage_form || 'N/A'}`);

  // Error information
  if (submission.error_message) {
    console.log('\n⚠️  ERROR INFORMATION:');
    console.log(`   Error Message: ${submission.error_message}`);
  }

  // Client review status
  console.log('\nClient Review Status:');
  console.log(`   Ready for MLR: ${submission.ready_for_mlr || false}`);
  console.log(`   Client Reviewed At: ${submission.client_reviewed_at || 'N/A'}`);
  console.log(`   Client Reviewed By: ${submission.client_reviewed_by || 'N/A'}`);

  // SEO review status
  console.log('\nSEO Review Status:');
  console.log(`   SEO Reviewed At: ${submission.seo_reviewed_at || 'N/A'}`);
  console.log(`   SEO Reviewed By: ${submission.seo_reviewed_by || 'N/A'}`);

  console.log('\n=== RAW DATA (for debugging) ===\n');
  console.log('All fields in submission:', Object.keys(submission).sort().join(', '));

  console.log('\n=== END OF DETAILED REPORT ===');
}

checkSubmissionDetailed().catch(console.error);